File: DOTNET-ANGULAR/backend/DTO/UserDTO.cs
using nfcbackend.Models;

namespace nfcbackend.DTO
{
    public class UserDTO : Base
    {
        public UserDTO()
        {
        }

        public string firstName { get; set; }

        public string lastName { get; set; }

        public string phoneNumber { get; set; }

        public string address { get; set; }

        public DateTime birth { get; set; }


        public AccountDTO? account { get; set; }

        public DateTime? CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

    }
}

File: DOTNET-ANGULAR/frontend/src/app/_interfaces/UserDTO.ts
import { AccountDTO } from "./AccountDTO";

export class UserDTO {
  firstName: string;
  lastName: string;
  phoneNumber: string;
  address: string;
  birth: string;
  account?: AccountDTO;
  createdAt?: string;
  updatedAt?: string;

  constructor(
    firstName: string,
    lastName: string,
    phoneNumber: string,
    address: string,
    birth: string,
    account?: AccountDTO,
    createdAt?: string,
    updatedAt?: string
  ) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.phoneNumber = phoneNumber;
    this.address = address;
    this.birth = birth;
    this.account = account;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
  }

}
